_data-prepper/pipelines/configuration/sources/s3.md

```diff
@@ -43,25 +84,70 @@ Option | Required | Type | Description
 :--- | :--- | :--- | :---
 region | No | String | The AWS Region to use for credentials. Defaults to [standard SDK behavior to determine the Region](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/region-selection.html).
 sts_role_arn | No | String | The AWS Security Token Service (AWS STS) role to assume for requests to Amazon SQS and Amazon S3. Defaults to null, which will use the [standard SDK behavior for credentials](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/credentials.html).
+aws_sts_header_overrides | No | Map | A map of header overrides to make when assuming the IAM role for the sink plugin.
 
+## codec
 
-## file
+The `codec` determines how the `s3` source parses each S3 object.
 
-Source for flat file input.
+### newline
 
-Option | Required | Type | Description
-:--- | :--- | :--- | :---
-path | Yes | String | The path to the input file (e.g. `logs/my-log.log`).
-format | No | String | The format of each line in the file. Valid options are `json` or `plain`. Default value is `plain`.
-record_type | No | String | The record type to store. Valid options are `string` or `event`. Default value is `string`. If you would like to use the file source for log analytics use cases like grok, set this option to `event`.
+The `newline` option parses each single line as a single log event. This is ideal for most application logs since each event parses per single line. It can also be suitable for S3 objects that have individual JSON objects on each line, which matches well when uses with the [parse_json]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/configuration/processors/parse-json/) processor to parse each line.
 
-## pipeline
+Option | Required | Type    | Description
+:--- | :--- |:--------| :---
+skip_lines | No | Integer | The number of lines to skip before creating events. You can use this configuration to skip common header rows. Defaults to `0`.
+header_destination | No | String  | A key value to assign the header line of the S3 object. If this option is specified then each Event will contain a header_destination field.
 
-Source for reading from another pipeline.
+### json
```

Comment:
```suggestion
### JSON
```

Reply:
Keeping this lowercase to match how the codec setting appears in the processor, which is `json`.

Reply:
Added `codec` for clarity here.